pkgs/swift2objc/lib/src/generator/_core/utils.dart

```diff
@@ -34,3 +37,14 @@ void outputNextToFile({
 
   File(outputPath).writeAsStringSync(content);
 }
+
+String generateAnnotations(Declaration decl) {
+  final annotations = StringBuffer();
+  if (decl is CanAsync && (decl as CanAsync).async) {
```

Comment:
Promotion isn't working here for some reason, so need the `as`.

Ideally I'd be able to do this with generics, like `String generateAnnotations<T extends CanAsync, CanThrow> (T decl) {`, but Dart doesn't allow that yet: https://github.com/dart-lang/language/issues/2709